src/modules/logger/logged_topics.cpp

```diff
@@ -177,6 +177,13 @@ void LoggedTopics::add_default_topics()
 	if (gps_dump_comm == 1) {
 		add_topic("gps_dump");
 	}
+
+	int32_t mav_dump_tunnel = 0;
+	param_get(param_find("MAV_DUMP_TUNNEL"), &mav_dump_tunnel);
```

Comment:
The analogous way to handle this is to have mavlink only republish the tunnel messages if `MAV_DUMP_TUNNEL` is enabled. Then on the logger side checking the param value before adding the topic is a small optimization, but not strictly required. In general try to keep modules contained to their own parameter space.

Reply:
Yes, that makes sense. The message is now translated according to the parameter.